src/u_websocket.c

```diff
@@ -69,19 +70,26 @@ static int is_websocket_data_available(struct _websocket_manager * websocket_man
 
 static ssize_t read_data_from_socket(struct _websocket_manager * websocket_manager, uint8_t * data, size_t len) {
   ssize_t ret = 0, data_len;
+  int retries = 0;
   
   if (len > 0) {
     do {
-      if (websocket_manager->tls) {
-        data_len = gnutls_record_recv(websocket_manager->gnutls_session, &data[ret], (len - ret));
-      } else if (websocket_manager->type == U_WEBSOCKET_SERVER) {
-        data_len = read(websocket_manager->mhd_sock, &data[ret], (len - ret));
-      } else {
-        data_len = read(websocket_manager->tcp_sock, &data[ret], (len - ret));
+      if (is_websocket_data_available(websocket_manager)) {
```

Comment:
This is the location of the additional poll action, I added a call to `is_websocket_data_available`, which performs the poll